torch/testing/_creation.py

```diff
@@ -123,8 +123,11 @@ def clamp(a, l, h):
         low = clamp(low, lowest, highest)
         high = clamp(high, lowest, highest)
 
-        if dtype in [torch.uint8, torch.int8, torch.int16, torch.int32, torch.int64]:
-            return math.floor(low), math.ceil(high)
+        if dtype in [torch.bool, torch.uint8, torch.int8, torch.int16, torch.int32, torch.int64]:
+            # 1. `low` is ceiled to avoid creating values smaller than `low` and thus outside the specified interval
+            # 2. Following the same reasoning as for 1., `high` should be floored. However, the higher bound of
+            #    `torch.randint` is exclusive and thus ceiling here as well is fine.
+            return math.ceil(low), math.ceil(high)
```

Comment:
I lean more towards `ceil/floor` than `ceil/ceil`, but this doesn't seem totally unreasonable

Reply:
I don't think `ceil(...), floor(...)` is correct. At the very least we need `ceil(...), floor(...) + 1`. Otherwise, `(- (1 - eps), 1 - eps)` would turn into `(0, 0)` and that doesn't work with `torch.randint`:

```pycon
>>> torch.randint(0, 0, ())
RuntimeError: random_ expects 'from' to be less than 'to', but got from=0 >= to=0
```

Both `ceil(...), ceil(...)` and `ceil(...), floor(...) + 1` work for the example above:

```pycon
>>> eps = 1e-4
>>> low, high = -(1 - eps), 2 - eps
>>> torch.manual_seed(0)
>>> torch.randint(math.ceil(low), math.floor(high) + 1, (10,))
tensor([0, 1, 1, 0, 1, 1, 1, 1, 1, 1])
>>> torch.manual_seed(0)
>>> torch.randint(math.ceil(low), math.ceil(high), (10,))
tensor([0, 1, 1, 0, 1, 1, 1, 1, 1, 1])
```

However, `ceil(...), floor(...) + 1` does not work if we give integer bounds, since we claim that `high` is exclusive:

https://github.com/pytorch/pytorch/blob/893aa5df3f2a475c91ea8eadb1353812e52fb227/torch/testing/_creation.py#L71

```pycon
>>> low, high = -1, 2
>>> torch.manual_seed(0)
>>> torch.randint(math.ceil(low), math.floor(high) + 1, (10,))
tensor([-1,  2,  0, -1,  2,  2,  2,  2,  0,  2])
>>> torch.manual_seed(0)
>>> torch.randint(math.ceil(low), math.ceil(high), (10,))
tensor([ 1, -1,  1, -1,  0, -1,  0,  0,  0, -1])
```

So it seems that `ceil(...), ceil(...)` is the only viable option without adding extra checks. Thoughts?

Reply:
There are plenty of test errors though. Let me investigate this a little further.

Reply:
Your analysis makes sense to me if high is exclusive